probe-rs/src/debug/variable.rs

```diff
@@ -412,11 +436,25 @@ impl Variable {
 
     /// Implementing get_value(), because Variable.value has to be private (a requirement of updating the value without overriding earlier values ... see set_value()).
     pub fn get_value(&self, variable_cache: &VariableCache) -> String {
-        if let Some(existing_value) = self.value.clone() {
-            // The `value` for this `Variable` is non empty because it is either:
-            // - A base data type for which a value was determined based on the core runtime
-            // - Contains an error that was encountered while resolving the `Variable`'s `DebugInfo`
+        #[allow(clippy::if_same_then_else)]
+        // because of limitations with ||'ing `if let` statements
+        if let Some(VariableError::IncludeAsChild(debug_error)) = self.variable_error.clone() {
+            // We encountered an error somewhere, so report it to the user
+            debug_error
+        } else if let Some(VariableError::RemoveFromParent(debug_error)) =
+            self.variable_error.clone()
+        {
+            // We encountered an error somewhere, so report it to the user
+            debug_error
```

Comment:
```suggestion
        } else if let Some(VariableError::RemoveFromParent(debug_error)) =
            &self.variable_error
        {
            // We encountered an error somewhere, so report it to the user
            debug_error.clone()
```

I'm not sure if this more efficient, but this is the more idiomatic way to write this, for me. So the clone only occurs if the value is actually needed, and not already in the comparison.

Reply:
Nice ... thank you :)